lighthouse-core/audits/webapp-install-banner.js

```diff
@@ -66,32 +67,37 @@ class WebappInstallBanner extends MultiCheckAudit {
   }
 
 
-  static assessServiceWorker(artifacts, failures) {
+  static assessServiceWorker(artifacts, result) {
     const hasServiceWorker = SWAudit.audit(artifacts).rawValue;
     if (!hasServiceWorker) {
-      failures.push('Site does not register a Service Worker');
+      result.failures.push('Site does not register a Service Worker');
     }
   }
 
-  static assessOfflineStartUrl(artifacts, failures) {
-    const hasOfflineStartUrl = artifacts.StartUrl === 200;
+  static assessOfflineStartUrl(artifacts, result) {
+    const hasOfflineStartUrl = artifacts.StartUrl.statusCode === 200;
+
     if (!hasOfflineStartUrl) {
-      failures.push('Manifest start_url is not cached by a Service Worker');
+      result.failures.push('Manifest start_url is not cached by a Service Worker');
+
+      if (artifacts.StartUrl.debugString) {
+        result.failures.push(artifacts.StartUrl.debugString);
+      }
+    } else if (artifacts.StartUrl.debugString) {
+      result.debugString = artifacts.StartUrl.debugString;
```

Comment:
weird, I left this comment before (and it was in the email), but it was eaten in the web interface:

> `MultiCheckAudit` is already somewhat confusing and this doesn't exactly simplify things :)

> Short of a `MultiCheckAudit` refactor, what about something like a `warnings` array, which is used for `debugString`, just like failures but they won't fail the audit?

> Just an idea, maybe we can come up with something simpler/more self-explanatory

so here (for this idea) it would `result.warnings.push(artifacts.StartUrl.debugString);` whether or not it `hasOfflineStartUrl`, and then `MultiCheckAudit` would handle appending `result.warnings.join(', ')` to the `debugString`